component/src/action_handler/actions/swap.rs

```diff
@@ -20,39 +20,17 @@ use crate::action_handler::ActionHandler;
 impl ActionHandler for Swap {
     #[instrument(name = "swap", skip(self, _context))]
     async fn check_stateless(&self, _context: Arc<Transaction>) -> Result<()> {
-        let swap = self;
-
-        // Check swap proof
-        swap.proof
-            .verify(
-                // TODO: no value commitments until flow encryption is available
-                // so we pass placeholder values here, the proof doesn't check these right now
-                // and will fail when checking is re-enabled.
-                Value {
-                    amount: 0u64.into(),
-                    asset_id: *STAKING_TOKEN_ASSET_ID,
-                }
-                .commit(Fr::zero()),
-                Value {
-                    amount: 0u64.into(),
-                    asset_id: *STAKING_TOKEN_ASSET_ID,
-                }
-                .commit(Fr::zero()),
-                swap.body.fee_commitment,
-                swap.body.swap_nft.note_commitment,
-                swap.body.swap_nft.ephemeral_key,
-            )
-            .context("A swap proof did not verify")?;
-
-        // TODO: are any other checks necessary?
+        self.proof.verify(
+            self.body.fee_commitment,
+            self.body.payload.commitment,
+            self.balance_commitment(),
+        )?;
```

Comment:
I streamlined the `Swap`/`SwapClaim` proofs, but there's still some followup work there (will note points)